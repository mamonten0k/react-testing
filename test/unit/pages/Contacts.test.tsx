import React from "react";
import { it } from "@jest/globals";
import { renderRoute, renderFragment } from "../render";

import { Contacts } from "../../../src/client/pages/Contacts";

describe("<Contacts>", () => {
  it("Представляет собой статичный контент", () => {
    const { asFragment } = renderFragment(<Contacts />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("По умолчанию присвоен роут /contacts", () => {
    const { container } = renderRoute("/contacts");
    expect(container.firstChild).toHaveClass("Contacts");
  });
});
