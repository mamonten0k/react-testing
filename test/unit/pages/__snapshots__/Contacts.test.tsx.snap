// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contacts> Представляет собой статичный контент 1`] = `
<DocumentFragment>
  <div
    class="Contacts"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1>
          Contacts
        </h1>
        <p>
          Ut non consequatur aperiam ex dolores. Voluptatum harum consequatur est totam. Aut voluptatum aliquid aut optio et ea. Quaerat et eligendi minus quasi. Culpa voluptatem voluptatem dolores molestiae aut quos iure. Repellat aperiam ut aliquam iure. Veritatis magnam quisquam et dolorum recusandae aut.
        </p>
        <p>
          Molestias inventore illum architecto placeat molestias ipsam facilis ab quo. Rem dolore cum qui est reprehenderit assumenda voluptatem nisi ipsa. Unde libero quidem. Excepturi maiores vel quia. Neque facilis nobis minus veniam id. Eum cum eveniet accusantium molestias voluptas aut totam laborum aut. Ea molestiae ullam et. Quis ea ipsa culpa eligendi ab sit ea error suscipit. Quia ea ut minus distinctio quam eveniet nihil. Aut voluptate numquam ipsa dolorem et quas nemo.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
